### PR TITLE
Docs: Reactive streams clarify cold publishers

### PR DESCRIPTION
**Pull request by rozza**
JAVA-3961

### CODE REVIEW

**Comment by rozza** [comment]
This is a difference to the pre reactor implementation. However, it's determined by how Mono's/Flux's work internally.

File: docs/reference/content/driver-reactive/getting-started/quick-start-primer.md [hunk 1]
```diff
@@ -49,12 +49,12 @@ where `T` is the type of response for the operation.
 
 {{% note %}}
 All [`Publishers`](http://www.reactive-streams.org/reactive-streams-1.0.1-javadoc/?org/reactivestreams/Publisher.html) returned 
-from the API are cold, meaning that no I/O happens until they are subscribed to and the subscription makes a request. So just creating a 
-`Publisher` won't cause any network IO. It's not until `Subscription.request()` is called that the driver executes the operation.
+from the API are cold, meaning that nothing happens until they are subscribed to. So just creating a
+`Publisher` won't cause any network IO, it's not until `Publisher.subscribe` is called that the driver executes the operation.
```
Comment (jyemin) [inline docs/reference/content/driver-reactive/getting-started/quick-start-primer.md:53]: Nit.  Two sentences is better grammar than a comma

File: docs/reference/content/driver-reactive/getting-started/quick-start-primer.md [hunk 2]
```diff
@@ -49,12 +49,12 @@ where `T` is the type of response for the operation.
...
 
 Publishers in this implementation are unicast. Each [`Subscription`](http://www.reactive-streams.org/reactive-streams-1.0.1-javadoc/?org/reactivestreams/Subscription.html) 
-to a `Publisher` relates to a single MongoDB operation and its ['Subscriber'](http://www.reactive-streams.org/reactive-streams-1.0.1-javadoc/?org/reactivestreams/Subscriber.html)  
-will receive its own specific set of results. 
+to a `Publisher` relates to a single MongoDB operation and its ['Subscriber'](http://www.reactive-streams.org/reactive-streams-1.0.
+1-javadoc/?org/reactivestreams/Subscriber.html) will receive its own specific set of results.
```
Comment (jyemin) [inline docs/reference/content/driver-reactive/getting-started/quick-start-primer.md:57]: I think the newline will cause problems

**Comment by stIncMale** [comment]
@rozza Thanks for doing the change!

**Review by stIncMale — APPROVED** [review]

File: docs/reference/content/driver-reactive/getting-started/quick-start-primer.md [hunk 3]
```diff
@@ -49,12 +49,12 @@ where `T` is the type of response for the operation.
 
 {{% note %}}
 All [`Publishers`](http://www.reactive-streams.org/reactive-streams-1.0.1-javadoc/?org/reactivestreams/Publisher.html) returned 
-from the API are cold, meaning that no I/O happens until they are subscribed to and the subscription makes a request. So just creating a 
-`Publisher` won't cause any network IO. It's not until `Subscription.request()` is called that the driver executes the operation.
+from the API are cold, meaning that nothing happens until they are subscribed to. So just creating a
```
Comment (stIncMale) [inline docs/reference/content/driver-reactive/getting-started/quick-start-primer.md:52]: Would it be helpful to link [cold](https://projectreactor.io/docs/core/release/reference/#reactor.hotCold)?

**Review by jyemin — CHANGES_REQUESTED** [review]

File: docs/reference/content/driver-reactive/getting-started/quick-start-primer.md (hunk 3 above)
Comment (rozza) [reply docs/reference/content/driver-reactive/getting-started/quick-start-primer.md]: Done

File: docs/reference/content/driver-reactive/getting-started/quick-start-primer.md (hunk 1 above)
Comment (rozza) [reply docs/reference/content/driver-reactive/getting-started/quick-start-primer.md]: Done

File: docs/reference/content/driver-reactive/getting-started/quick-start-primer.md (hunk 2 above)
Comment (rozza) [reply docs/reference/content/driver-reactive/getting-started/quick-start-primer.md]: Fixed

**Review by jyemin — APPROVED** [review]
LGTM

**Pull request merged**